### DISCUSSION TITLE
Mendukung WhatsApp Multi-Device Beta

### DISCUSSION THREAD

**Discussion by rioastamal** [kirimwa-id/forum#42] [Roadmap]
Saat ini WhatsApp sudah meluncurkan WhatsApp Beta Multi-Device dimana maksimal dapat terhubung ke 4 perangkat sekaligus dan HP tidak perlu selalu terhubung ke internet.

API KirimWA.id belum mendukung fitur WhatsApp Multi-Device Beta. Namun kedepan fitur ini rencananya akan segera didukung secepat mungkin ketika WhatsApp Multi-Device Beta sudah memasuki rilis GA (General Availability).

Referensi:
https://faq.whatsapp.com/general/download-and-installation/about-multi-device-beta/?lang=en

**Comment by kafaa-dev** [comment]
Jika API KirimWA.id sudah mendukung fitur Whatsapp Multi-Device Beta, apakah nanti kedepannya akan bisa menggunakan Whatsapp Web? Di halaman [https://developer.kirimwa.id/#faq](https://developer.kirimwa.id/#faq) tertulis bahwa sebaiknya JANGAN menggunakan Whatsapp Web, karena koneksi WhatsApp di server api.kirimwa.id akan terputus. Tapi, karena Multi-Device, bukankah seharusnya masih bisa, dan tidak mengganggu proses pengiriman pesan di API KirimWA.id?

Saya akan menunggu jawabannya 🙏

**Reply by rioastamal** [reply]
@syahid246,

Benar, ketika Multi Device sudah diimplementasikan maka secara teori WhatsApp Web masih bisa digunakan karena setiap linked device akan memiliki `state` sendiri-sendiri ketika out-of-sync. 

FAQ tentu akan diubah seiring dengan berubahnya fitur.

**Reply by huiralb** [reply]
FYI, whatsapp membatasi max 5 device.

**Comment by shaviscore** [comment]
Sunggu Menarik jika sudah bisa Multi Device saya akan mencoba nya

**Comment by zainul** [comment]
Mas @rioastamal mau tanya
seumpama di HP saya dan no yg saya pakai sudah update ke Whatsapp Beta apakah tidak akan bisa juga, bagaimana mengatasi case ini apakah harus mengganti no baru ?

**Reply by kafaa-dev** [reply]
Bukankah di WhatsApp ada tombol untuk menonaktifkan WhatsApp Beta? Ada di menu **Tautkan Perangkat**/**Connected Device**.

**Reply by rioastamal** [reply]
Betul kata mas @syahid246, pada halaman "Linked Devices" harusnya ada tombol "Leave Beta". Jadi bis on/off kapan saja. Namun ketika on/off maka harus re-scan ulang karena WhatsApp akan menghapus session pada device.

**Comment by baguwes** [comment]
sementara untuk kebutuhan development masih harus menggunakan HP yang berbeda atau membuat duplicate whatsapp dengan nomor lain, semoga  multi-device login bisa cepat terlaksana.

semangat team KirimWA.

**Reply by rioastamal** [reply]
Terima kasih untuk dukungannya 👍

**Comment by azmirizkifar20** [comment]
mau tanya mas @rioastamal, kalau nanti API KirimWA.id sudah mendukung fitur WhatsApp Multi-Device Beta, apakah nnti bakal support perangkat yang selalu online selama wa-web nya masih berjalan? karena saat ini kan kita bisa menggunakan wa-web walaupun kondisi hp sedang tidak berada di jaringan internet. apakah nanti API KirimWA.id juga akan support?

terimakasih sebelumnya,

**Reply by kafaa-dev** [reply]
wah iya juga, apa nggak harus reconnect lagi nanti ya :D

**Reply by rioastamal** [reply]
@azmirizkifar20 , Seperti yang dituliskan pada [website resmi](https://faq.whatsapp.com/general/download-and-installation/about-multi-device-beta/?lang=en) WhatsApp bahwa Multi-device beta dapat mendukung hingga 4 device. Jadi ketika nanti API KirimWA.id sudah mendukung Multi-Device maka API KirimWA.id tetap dapat digunakan meskipun device WhatsApp dalam kondisi tanpa internet.

Karena maksimal 4 device maka selain API KirimWA.id maka WhatsApp juga harusnya tetap dapat digunakan bersamaan dan tidak perlu juga internet pada device.

**Comment by Mastrios** [comment]
Apakah kirimWa.id sudah mendukung fitru multi device ?

**Reply by rioastamal** [reply]
Belum. Silahkan subscribe thread [Pengumuman](https://github.com/kirimwa-id/forum/discussions/categories/pengumuman) untuk mendapatkan notifikasi ketika ada rilis terbaru.

**Comment by nubinter** [comment]
Hai, saya pengguna baru kirimwa
permasalahan saya adalah setiap saya ingin scan qrcode untuk menghubungkan akun wa saya, selalu gagal. Apakah karena aplikasi wa yang saya gunakan adalah versi terbaru? saya instal langsung di playstore dan tidak tergabung ke program beta. Tapi pada menu linked device sepertinya sudah langsung mendukung multi device. apa solusinya ya?

**Reply by rioastamal** [reply]
API KirimWA.id masih belum mendukung Multi-Device. Saya belum bisa menentukan kapan rilis untuk dukungan multi-device, namun yang jelas kedepan fitur ini akan didukung.

**Reply by nubinter** [reply]
Solusinya bagaimana ya? Saya sufah coba install whatsapp versi lama di second space tapi masih gagal juga. Mohon bantuannya mas @rioastamal .
Versi whatsapp yang sy gunakan versi 2.22.7.74

**Reply by kafaa-dev** [reply]
Apakah sudah mencoba di reconnect device nya?

**Reply by nubinter** [reply]
@syahid246 belum pernah berhasil connect mas. Selalu gagal, disuruh scan ulang terus. Apakah wa-nya harus pake hp jadul ya?

**Reply by rioastamal** [reply]
Jika versi dari WA adalah multi-device maka belum bisa menggunakan API KirimWA.id. Solusinya downgrade versi WA atau tunggu rilis dukungan untuk multi-device.

**Comment by shahib-mirbath** [comment]
Semoga segera terealisasi. Amin

**Reply by rioastamal** [reply]
@shahib-mirbath jangan lupa subscribe thread ini sehingga mendapat notifikasi ketika rilis dilakukan.

**Reply by shahib-mirbath** [reply]
Sudah doong.. hehe

**Comment by rioastamal** [comment]
Buat rekan-rekan semua, silahkan watch repository ini dan juga subscribe thread ini agar mendapat notifikasi ketika rilis dilakukan. Fitur ini masih dikerjakan :)

**Reply by shahib-mirbath** [reply]
Siap..

**Reply by Bisbiss** [reply]
Siap mas, mantab

**Reply by melodiazz** [reply]
semangat pak @rioastamal

**Reply by gustysap** [reply]
Wuhuuu ditunggu banget , mantap mas @rioastamal

**Reply by shahib-mirbath** [reply]
Semangaaat @rioastamal 
Device saya sudah tdk ada pilihan 😅

**Reply by rioastamal** [reply]
@shahib-mirbath hehe mohon bersabar dulu 😄

**Reply by shahib-mirbath** [reply]
@rioastamal Siiaap.. hehee

**Reply by favianriyanto** [reply]
apa kabar pak? :v @rioastamal

**Reply by baguwes** [reply]
semangat mas, sayangnya saya belum expert jadi gak bisa bantu-bantu :upside_down_face:

**Reply by kafaa-dev** [reply]
semangat mas :)

**Reply by shahib-mirbath** [reply]
@rioastamal semoga diberi kelancaran. Aamiiiin

**Comment by newfaboc** [comment]
Apa ada versi whatsapp yang bisa dipakai lagi kah ? sebab saya coba pakai versi 2.2.22.73 sudah harus update yang baru.

Terima kasih sebelumnya.

**Reply by ekoyuliawan** [reply]
Pengalaman saya, sudah tidak bisa, baik yang official maupun mod. Semua versi akan force update ke versi multi-device. Semoga yang multi-device segera rilis. Aamiiin.

**Comment by rioastamal** [comment]
UPDATE:
Sedang mencoba versi multi device API KirimWA v1.0-RC14

![Screen Shot 2022-07-16 at 13 28 05](https://user-images.githubusercontent.com/469847/179343201-84b19cd6-c6c1-4175-9ae7-69d34cb4d177.png)

**Reply by kafaa-dev** [reply]
Mantap mas 👍

**Reply by yudha2003** [reply]
Ditunggu mas rio 😁

**Reply by sanusi86** [reply]
Semangat mas rio

**Reply by kotak47** [reply]
mantap mas..

**Comment by newfaboc** [comment]
untuk scan device masih belum bisa ya mas rio ?

**Reply by rioastamal** [reply]
Fitur multi device belum dirilis. Diatas hanya sebuah tes di local mesin saya.

**Comment by rioastamal** [comment]
Kabar gembira 🎉  fitur multi device telah dirilis. Simak pengumumannya di thread #120 . Selamat mencoba!!

**Reply by shahib-mirbath** [reply]
Alhamdulillaaaah

**Reply by shahib-mirbath** [reply]
terimakasih mas @rioastamal
